ntoskrnl/se/debug.c

```diff
@@ -0,0 +1,479 @@
+/*
+ * PROJECT:         ReactOS Kernel
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Security subsystem debug routines support
+ * COPYRIGHT:       Copyright 2022 George Bișoc <george.bisoc@reactos.org>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* PRIVATE FUNCTIONS **********************************************************/
+
+/**
+ * @brief
+ * Converts an Access Control Entry (ACE) type
+ * to a string.
+ *
+ * @param[in] AceType
+ * The type of ACE to be given.
+ *
+ * @return
+ * Returns a converted ACE type strings. If no
+ * known ACE type is found, it will return
+ * UNKNOWN TYPE.
+ */
+static
+PCHAR
+SepGetAceTypeString(
+    _In_ UCHAR AceType)
+{
+    switch (AceType)
+    {
+        case ACCESS_DENIED_ACE_TYPE:
+        {
+            return "ACCESS_DENIED_ACE_TYPE";
+        }
+
+        case ACCESS_ALLOWED_ACE_TYPE:
+        {
+            return "ACCESS_ALLOWED_ACE_TYPE";
+        }
+
+        case ACCESS_DENIED_OBJECT_ACE_TYPE:
+        {
+            return "ACCESS_DENIED_OBJECT_ACE_TYPE";
+        }
+
+        case ACCESS_ALLOWED_OBJECT_ACE_TYPE:
+        {
+            return "ACCESS_ALLOWED_OBJECT_ACE_TYPE";
+        }
+
+        case SYSTEM_AUDIT_ACE_TYPE:
+        {
+            return "SYSTEM_AUDIT_ACE_TYPE";
+        }
+
+        case SYSTEM_ALARM_ACE_TYPE:
+        {
+            return "SYSTEM_ALARM_ACE_TYPE";
+        }
+
+        default:
+            return "UNKNOWN TYPE";
+    }
+}
+
+/**
+ * @brief
+ * Dumps the ACE flags to the debugger
+ * output.
```

Comment:
why the line-break after the word debugger? Plenty of characters left in the line.